src/backends/r1cs/mod.rs

```diff
@@ -374,6 +390,11 @@ where
 
         // check if every cell vars end up being a cell var in the circuit or public output
         for (index, _) in self.witness_vars.iter().enumerate() {
+            // skip the first var that is assumed to be the factor of constants of all linear combinations
```

Comment:
wdym "the factor of constants of all linear combinations"?

Reply:
The first var is assumed to be the coefficient of the constant of a linear combination.

In a linear combination, each of the vars can be paired with a coefficient.
The constant acts as the coefficient while var part is the always the first var (1). Look at it another way, the first var can always be the factor of the constants.

Reply:
don't tell me, tell it in the comment :) I was just pointing out that the comment was not clear